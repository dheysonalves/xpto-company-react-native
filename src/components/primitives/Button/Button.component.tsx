import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import { Text } from "../Themed";

interface IButtonProps {
	text: string;
	containerBg?: string;
	handleSubmit: () => void;
}

const Button = ({
	containerBg,
	text = "submit",
	handleSubmit,
	...otherProps
}: IButtonProps & TouchableOpacity["props"]) => {
	const bg = containerBg ? containerBg : "#000";
	return (
		<TouchableOpacity
			style={[styles.button, { backgroundColor: bg }]}
			activeOpacity={0.8}
			onPress={handleSubmit}
			{...otherProps}>
			<Text style={styles.text}>{text}</Text>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	button: {
		alignItems: "center",
		justifyContent: "center",
		height: 50,
		borderRadius: 4,
	},
	text: {
		color: "#fff",
		textTransform: "uppercase",
		fontSize: 16,
		fontWeight: "bold",
	},
});

export default Button;
