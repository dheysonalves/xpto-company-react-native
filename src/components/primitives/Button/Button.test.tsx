import React from "react";

import { render, cleanup } from "@testing-library/react-native";
import Button from "./Button.component";
import renderer from "react-test-renderer";

const ButtonRender = () => (
	<Button
		text="SUBMIT"
		handleSubmit={() => {}}
		testID="pressMeButton"
		accessibilityLabel="Press me"
	/>
);

afterEach(cleanup);

describe("Button Component Test Suite", () => {
	it("should render the Button", () => {
		const tree = renderer.create(<ButtonRender />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it("should find the Button testID", () => {
		const testIdName = "pressMeButton";

		const { getByTestId } = render(<ButtonRender />);

		const foundButton = getByTestId(testIdName);

		expect(foundButton).toBeTruthy();
	});

	it("should find the Button via accessibilityLabel", () => {
		const accessibilityLabel = "Press me";

		const { getByA11yLabel } = render(<ButtonRender />);

		const foundButton = getByA11yLabel(accessibilityLabel);

		expect(foundButton).toBeTruthy();
	});

	it("should find the Button title", () => {
		const title = "SUBMIT";

		const { getByText } = render(<ButtonRender />);

		const foundButtonTitle = getByText(title);

		expect(foundButtonTitle.props.children).toEqual(title);
	});
});
