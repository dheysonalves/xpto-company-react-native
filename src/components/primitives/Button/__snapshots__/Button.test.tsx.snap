// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Test Suite should render the Button 1`] = `
<View
  accessibilityLabel="Press me"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderRadius": 4,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="pressMeButton"
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
          "textTransform": "uppercase",
        },
      ]
    }
  >
    SUBMIT
  </Text>
</View>
`;
